### PR TITLE
Add VQFN-32-1EP_5x5mm_P0.5mm_EP3.5x3.5mm

### PR DESCRIPTION
**Pull request by jonas-schievink**
Used by FTDI's FT4222H.

Based on the data in https://www.ftdichip.com/Support/Documents/TechnicalNotes/TN_166%20FTDI%20Example%20IC%20PCB%20Footprints.pdf#page=22

I had to fight the script a little, because it would automatically make the footprint larger than what I entered, but the values are based off the TN I linked, so they should be kept as-is. The leads are still longer than what the TN specifies (0.8mm instead of 0.75mm).

Symbol PR: https://github.com/KiCad/kicad-symbols/pull/2091

Footprint PR: https://github.com/KiCad/kicad-footprints/pull/1786

### CODE REVIEW

**Comment by poeschlr** [comment]
The linked document does not contain the right data. This generator does not take a suggested footprint as input but the dimensions (including tolerances) of the packages itself. Meaning it takes the input as found in https://www.ftdichip.com/Support/Documents/DataSheets/ICs/DS_FT4222H.pdf#page=40

**Comment by jonas-schievink** [comment]
@poeschlr That makes sense. Fixed the dimensions (only the lead length was off).

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 1]
```diff
@@ -560,3 +560,62 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
...
+  thermal_vias:
+    count: [3, 3]
+    drill: 0.33
+    # min_annular_ring: 0.15
+    paste_via_clearance: 0.1
+    # EP_num_paste_pads: [2, 2]
+    # paste_between_vias: 1
+    # paste_rings_outside: 1
+    EP_paste_coverage: 0.7
+    grid: [1.2, 1.2]
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: Is there a reason to manually enter the grid?

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 2]
```diff
@@ -560,3 +560,62 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
...
+  EP_size_y:
+    minimum: 3.45
+    nominal: 3.5
+    maximum: 3.55
+  # EP_paste_coverage: 0.65
+  EP_num_paste_pads: [2, 2]
+
+  thermal_vias:
+    count: [3, 3]
+    drill: 0.33
```
[the remark this answers is not in the document]
Comment (jonas-schievink) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: No, this is just copied from the footprint above. The datasheet doesn't mention thermal vias at all, so theoretically I could just remove them completely.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml (hunk 1 above)
Comment (jonas-schievink) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: No, this is also due to copy-pasting the footprint above.

Comment (poeschlr) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: In this case remove the grid definition.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (poeschlr) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: In this case i would simply use 0.3mm (The same package can be used by a part that requires them so it makes sense to have this footprint)

[the remark this answers is not in the document]
Comment (jonas-schievink) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: Okay, done that.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml (hunk 1 above)
Comment (jonas-schievink) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: Also removed this. The FP now looks like this:

screenshot-2019-09-01-22:18:19

**Comment by poeschlr** [comment]
thanks

**Pull request merged**